src/app/GitUI/RepositoryCurrentBranchNameProvider.cs

```diff
@@ -26,6 +26,12 @@ public string GetCurrentBranchName(string repositoryPath)
         }
 
         string branchName = GitModule.GetSelectedBranchFast(repositoryPath);
+
+        if (string.IsNullOrWhiteSpace(branchName))
+        {
+            branchName = new GitModule(repositoryPath).GetSelectedBranch();
```

Comment:
`GetSelectedBranch` starts with invoking `GetSelectedBranchFast`:
https://github.com/gitextensions/gitextensions/blob/9dd29b2edefafc9c6a46861895776696dae0b1eb/src/app/GitCommands/Git/GitModule.cs#L2900

I don't think this is the most optimal way of addressing the issue. Instead, we should pass `Func<IGitModule>` into the .ctor (the same way we do in many other places), and then call
```cs
string branchName = getModule().GetSelectedBranch();
```

`GetSelectedBranchFast` should be turned into a private (or internal for testing purposes) method.

Reply:
> GetSelectedBranchFast should be turned into a private (or internal for testing purposes) method.

GetSelectedBranchFast is used in other modules when the branch name is not critical (for display) and speed is important, so it should remain public.